modules/packages/LinearAlgebra.chpl

```diff
@@ -1243,21 +1247,22 @@ proc trace(A: [?D] ?eltType) {
   return trace;
 }
 
-private proc _lu (in A: [?Adom] ?eltType) {
+/* LU helper function */
+private proc _lu(in A: [?Adom] ?eltType) {
   const n = Adom.shape(0);
-  const LUDom = {1..n, 1..n};
+  const dim = 0..<n;
+  const LUDom = {dim, dim};
```

Comment:
Just noting that this contradicts with the "inherit the domain from the input" philosophy. But it used to contradict in favor of 1-based domains, too.

It may be non-trivial to follow that philosophy here given the rest of the implementation here.

Reply:
Yes, unfortunately several functions contradict this philosophy. I tried to cover this in the PR message:

> A test used a function that assumed 1-based indexing and now assumes 0-based indexing, so was updated to assume 0-based indexing.

+

> Some of these updates revealed that some functions assumed 1-based indices. In this PR, we've updated them to assume 0-based indices instead. Future work should make all linear algebra functions index-neutral.

I think we should make a follow-up effort to:


1) beef up linear algebra testing to lock in this philosophy on all functions
2) fix any functions where this philosophy is not respected. This may just involve a bunch of `reindex` calls within, though we could try more optimized approaches down the road (given `reindex` introduces some overhead).